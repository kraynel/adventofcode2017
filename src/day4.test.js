const {
  countPassphrase,
  countPassphrase2,
  isValid,
  isValid2
} = require("./day4");
const fs = require("fs");

describe("countPassphrase", () => {
  it("should check if is valid 1", () => {
    expect(isValid("aa bb cc dd ee")).toBeTruthy();
  });

  it("should check if is valid 2", () => {
    expect(isValid("aa bb cc dd aa")).toBeFalsy();
  });

  it("should check if is valid 3", () => {
    expect(isValid("aa bb cc dd aaa")).toBeTruthy();
  });

  it("should count correct passphrases", () => {
    expect(
      countPassphrase(`aa bb cc dd aaa
aa bb cc dd ee
aa bb cc dd aa`)
    ).toBe(2);
  });

  it("should log the correct value", done => {
    fs.readFile("inputs/day4.txt", "utf8", (err, data) => {
      expect(countPassphrase(data)).toMatchSnapshot();
      done();
    });
  });
});

describe("countPassphrase2", () => {
  it("should check if is valid 1", () => {
    expect(isValid2("abcde fghij")).toBeTruthy();
  });

  it("should check if is valid 2", () => {
    expect(isValid2("abcde xyz ecdab")).toBeFalsy();
  });

  it("should check if is valid 3", () => {
    expect(isValid2("a ab abc abd abf abj")).toBeTruthy();
  });

  it("should check if is valid 4", () => {
    expect(isValid2("iiii oiii ooii oooi oooo")).toBeTruthy();
  });

  it("should check if is valid 5", () => {
    expect(isValid2("oiii ioii iioi iiio")).toBeFalsy();
  });

  it("should log the correct value", done => {
    fs.readFile("inputs/day4.txt", "utf8", (err, data) => {
      expect(countPassphrase2(data)).toMatchSnapshot();
      done();
    });
  });
});
