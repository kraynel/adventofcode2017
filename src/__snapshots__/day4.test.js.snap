// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countPassphrase should log the correct value 1`] = `466`;

exports[`countPassphrase2 should log the correct value 1`] = `251`;
